internal/configs/configurator.go

```diff
@@ -996,8 +1006,21 @@ func (cnf *Configurator) updatePlusEndpoints(ingEx *IngressEx) error {
 	return nil
 }
 
+// EnableReloads enables NGINX reloads meaning that configuration changes will be followed by a reload.
+func (cnf *Configurator) EnableReloads() {
+	cnf.isReloadsEnabled = true
+}
+
+func (cnf *Configurator) reload(isEndpointsUpdate bool) error {
+	if !cnf.isReloadsEnabled {
+		return nil
+	}
+
+	return cnf.nginxManager.Reload(isEndpointsUpdate)
+}
+
 // UpdateConfig updates NGINX configuration parameters.
-func (cnf *Configurator) UpdateConfig(cfgParams *ConfigParams, ingExes []*IngressEx, mergeableIngs []*MergeableIngresses, virtualServerExes []*VirtualServerEx) (Warnings, error) {
+func (cnf *Configurator) UpdateConfig(cfgParams *ConfigParams, resources ExtendedResources) (Warnings, error) {
```

Comment:
Should we address this lint failure now while we're in this part of the code?

Reply:
better to do it separately

Reply:
Isn't that just creating technical debt? Was this lint error here before the changes?

Reply:
yep, it was there before. the linter is configured so that it doesn't show the errors for the existing code, only for the new code. the new code in the PR changed the existing code with the existing problem, so that's why the linter complained.
better to stay focus in this PR and not increase its scope any further, and acknowledge and address the problematic method (there are more than one in Configurator with high complexity) separately